third-party/build.sh

```diff
@@ -65,67 +97,87 @@ fi
-OS=`grep -m1 "ID=" /etc/os-release | sed 's/"//g' | sed 's/ID=//g' `
+OS=`grep -m1 "^ID=" /etc/os-release | sed 's/"//g' | sed 's/ID=//g' `
 
 case $OS in
         "ubuntu")
-
-		sudo apt-get install --assume-yes build-essential autoconf libtool pkg-config
-		if [[ $? > 0 ]]
-		then
-			echo "apt-get install failed, exiting..."
-			exit 1
-		fi
 		case $(get_ubuntu_release) in
-		    *16.04*)
-			sudo apt-get install --assume-yes libpthread-stubs0-dev libevent-dev flex hwloc libhwloc-dev libhwloc-plugins libyaml-cpp-dev python python3 python-pip python3-pip
+		     *20.04*)
+			if [ "$no_package_install" != "true" ]
+                        then
+			        sudo apt-get install --assume-yes ${ubuntu20_package_list}
+		                if [[ $? > 0 ]]
+		                then
+			            echo "apt-get install failed, exiting..."
+		   	            exit 1
+		                fi
+			fi
+	    		;;		    
+    		    *18.04*)
+			if [ "$no_package_install" != "true" ]
+			then	
+			    sudo apt-get install --assume-yes ${ubuntu18_package_list}
+		            if [[ $? > 0 ]]
+      	                    then
+			        echo "apt-get install failed, exiting..."
+			        exit 1
+		            fi
+
+			fi
 			;;
-		    *)
-			sudo apt-get install --assume-yes libpthread-stubs0-dev libevent-2.1-6 libevent-dev flex hwloc libhwloc-dev libhwloc-plugins libyaml-cpp-dev python python3 python-pip python3-pip
 		esac
-		if [[ $? > 0 ]]
-		then
-			echo "apt-get install failed, exiting..."
-			exit 1
-		fi
-
 		;;
 	"rhel" | "centos")
-		sudo yum install --assumeyes gcc gcc-c++ kernel-devel make cmake autoconf-archive glibc python python-devel python3 automake libtool doxygen epel-release python-pip python3-pip
-		sudo yum install --assumeyes libevent cmake xmlto libevent-devel yaml-cpp yaml-cpp-devel
-		if [[ $? > 0 ]]
-		then
-			exit 1
-		fi
-
-		echo "Installing boost-1.63.0"
-		if [[ ! -e boost_1_63_0 ]]; then
-		    wget https://sourceforge.net/projects/boost/files/boost/1.63.0/boost_1_63_0.tar.gz
-		    tar -zxvf boost_1_63_0.tar.gz
-		fi
-
-		cd boost_1_63_0/
-		./bootstrap.sh
-		./b2 install --prefix=$ABS_BUILD_DIR
-		if [[ $? > 0 ]]
+		ver='rpm -E %{rhel}'
+		case $(ver) in
+			8)
+			 if [ "$no_package_install" != "true" ]
+			 then
+				sudo yum install --assumeyes ${rhel8_package_list}
+				if [[ $? > 0 ]]
+				then
+					echo "yum install failed, exiting..."
+					exit 1
+				fi
+			 fi
+		         ;;
+	                 esac
+			;;
+	"opensuse-leap" | "sles")
+		#TODO: Add version check
+		if [ "$no_package_install" != "true" ]
 		then
+                    sudo zypper --non-interactive install ${sles_package_list}
+		    if [[ $? > 0 ]]
+		    then
+			echo "zypper install failed, exiting..."
 			exit 1
-		fi
-		export BOOST_ROOT=$ABS_BUILD_DIR
-		;;
+		    fi 
+	        fi
+	        ;;
 
 esac		
-
-pip3 install ruamel.yaml
+if [ "$no_package_install" != "true" ] 
+then
+    pip3 install ruamel.yaml
+fi
 
 echo "Finished installing required RPMS"
```

Comment:
Message is required only if  [ "$no_package_install" != "true" ] . So push it within the if conditional after pip3 install